ansible_base/rbac/models/role.py

```diff
@@ -113,18 +113,39 @@ def give_creator_permissions(self, user, obj) -> Optional['RoleUserAssignment']:
             maybe_reverse_sync_assignment(assignment)
             return assignment
 
+    def _find_by_permissions(self, permissions):
+        """Find RoleDefinition matching the given permission set."""
+        for existing_rd in self.prefetch_related('permissions'):
+            existing_set = {perm.codename for perm in existing_rd.permissions.all()}
+            if existing_set == permissions:
+                return existing_rd
+        return None
+
     def get_or_create(self, permissions=(), defaults=None, **kwargs):
-        "Add extra feature on top of existing get_or_create to use permissions list"
+        """Add extra feature on top of existing get_or_create to use permissions list.
+
+        Handles race conditions where multiple threads may try to create the same
+        RoleDefinition simultaneously by catching IntegrityError and retrying the lookup.
+        """
         if permissions:
             permissions = set(permissions)
-            for existing_rd in self.prefetch_related('permissions'):
-                existing_set = {perm.codename for perm in existing_rd.permissions.all()}
-                if existing_set == permissions:
-                    return (existing_rd, False)
+            existing = self._find_by_permissions(permissions)
+            if existing:
+                return (existing, False)
+
             create_kwargs = kwargs.copy()
             if defaults:
                 create_kwargs.update(defaults)
-            return (self.create_from_permissions(permissions=permissions, **create_kwargs), True)
+            try:
+                with transaction.atomic():
+                    return (self.create_from_permissions(permissions=permissions, **create_kwargs), True)
+            except IntegrityError:
+                # Race condition: another thread created the RoleDefinition between
+                # our check and create. Re-query to find the one that was just created.
+                existing = self._find_by_permissions(permissions)
```

Comment:
Add a log here for debugging please. This is out of the ordinary, and `logger.info` would be appropriate in my book, and including `self.pk` in the message would be important.